From c5dcf6c13db3cb9ff8de9c59e3c197f4b42a43b6 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Mon, 9 Aug 2021 14:13:51 +0200
Subject: [PATCH 3/6] acp: compare the HW volume against stored HW volume

When we store the real_volume we get a hardware_volume as stored
in the mixer and a residual software_volume.

When we read the volume from the card, we need to compare this against
the hardware_volume we stored to check if something changed, not
against the real_volume that also contains the leftover software_volume.
---
 spa/plugins/alsa/acp/acp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/spa/plugins/alsa/acp/acp.c b/spa/plugins/alsa/acp/acp.c
index 412588cb6..1ff1c5f95 100644
--- a/spa/plugins/alsa/acp/acp.c
+++ b/spa/plugins/alsa/acp/acp.c
@@ -1047,10 +1047,10 @@ static int read_volume(pa_alsa_device *dev)
 	/* Shift down by the base volume, so that 0dB becomes maximum volume */
 	pa_sw_cvolume_multiply_scalar(&r, &r, dev->base_volume);
 
-	if (pa_cvolume_equal(&dev->real_volume, &r))
+	if (pa_cvolume_equal(&dev->hardware_volume, &r))
 		return 0;
 
-	dev->real_volume = r;
+	dev->real_volume = dev->hardware_volume = r;
 
 	pa_log_info("New hardware volume: min:%d max:%d",
 			pa_cvolume_min(&r), pa_cvolume_max(&r));
-- 
2.31.1

